src/downloadAzureProject/setupProjectFolder.ts

```diff
@@ -0,0 +1,122 @@
+import { WebSiteManagementModels } from '@azure/arm-appservice';
+import * as extract from 'extract-zip';
+import { URL } from 'url';
+import * as vscode from 'vscode';
+import { callWithTelemetryAndErrorHandling, IActionContext } from 'vscode-azureextensionui';
+import { ProjectLanguage } from '../constants';
+import { GlobalStates } from '../extension';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { getNameFromId } from '../utils/azure';
+import { requestUtils } from '../utils/requestUtils';
+
+interface ILocalDevelopmentRequiredInputs {
+    resourceId: string | null;
+    defaultHostName: string | null;
+    devContainerName: string | null;
+    language: string | null;
+    downloadAppContent: string | null;
+}
+
+export async function setupProjectFolder(uri: vscode.Uri, filePath: string): Promise<void> {
+    const requiredInputs: ILocalDevelopmentRequiredInputs = getAllRequiredInputs(uri.query);
+    const resourceId: string | null = requiredInputs.resourceId;
+    const defaultHostName: string | null = requiredInputs.defaultHostName;
```

Comment:
You shouldn't need `defaultHostName` as long as we have `resourceId`. You can get it from the node.
```
slotTreeItem.client.defaultHostUrl
```